tmva/sofie/src/RModel_GNN.cxx

```diff
@@ -0,0 +1,235 @@
+#include <limits>
+#include <algorithm>
+#include <cctype>
+
+#include "TMVA/RModel_GNN.hxx"
+
+
+namespace TMVA{
+namespace Experimental{
+namespace SOFIE{
+
+
+    RModel_GNN::RModel_GNN(RModel_GNN&& other){
+      edges_update_block = std::move(other.edges_update_block);
+      nodes_update_block = std::move(other.nodes_update_block);
+      globals_update_block = std::move(other.globals_update_block);
+
+      edge_node_agg_block = std::move(other.edge_node_agg_block);
+      edge_global_agg_block = std::move(other.edge_global_agg_block);
+      node_global_agg_block = std::move(other.node_global_agg_block);
+
+      num_nodes = std::move(other.num_nodes);
+      num_edges = std::move(other.num_edges);
+      senders = std::move(other.senders);
+      receivers = std::move(other.receivers);
+
+      fName = std::move(other.fName);
+      fFileName = std::move(other.fFileName);
+      fParseTime = std::move(other.fParseTime);
+    }
+
+   RModel_GNN& RModel_GNN::operator=(RModel_GNN&& other){
+      edges_update_block = std::move(other.edges_update_block);
+      nodes_update_block = std::move(other.nodes_update_block);
+      globals_update_block = std::move(other.globals_update_block);
+
+      edge_node_agg_block = std::move(other.edge_node_agg_block);
+      edge_global_agg_block = std::move(other.edge_global_agg_block);
+      node_global_agg_block = std::move(other.node_global_agg_block);
+
+      num_nodes = std::move(other.num_nodes);
+      num_edges = std::move(other.num_edges);
+      senders = std::move(other.senders);
+      receivers = std::move(other.receivers);
+
+      fName = std::move(other.fName);
+      fFileName = std::move(other.fFileName);
+      fParseTime = std::move(other.fParseTime);
+
+      return *this;
+    }
+
+    RModel_GNN::RModel_GNN(const GNN_Init& graph_input_struct){
+        edges_update_block.reset((graph_input_struct.edges_update_block).get());
```

Comment:
You are converting here from a shared_ptr from `GNN_Init::edges_update_block` to a unique_ptr. You should call `reset` on the shared_ptr. But why the data members of `GNN_Init` are shared_ptr and not unique_ptr ?

Reply:
Thanks for that review, I have fixed that now.